File: fizzbuzz/C++/fizzbuzz.cpp
#include <iostream>
#include "gtest/gtest.h" 
using namespace std;

//For loop based Fizzbuzz
int main(int argc, char **argv) {
  for (int i = 1; i <= 100; i++){
    if (i % 5 == 0 && i % 3 == 0) {
      cout << "FizzBuzz\n";
    }
    else if (i % 3 == 0) {
      cout << "Fizz\n";
    }  
    else if (i % 5 == 0) {
      cout << "Buzz\n";
    }
    else {
      cout << i << "\n";
    }
  }
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();

}

//Function based Fizzbuzz created to use Gtest

string fizzbuzz(int i){
  if (i % 5 == 0 && i % 3 == 0) {
      return "FizzBuzz";
    }
    else if (i % 3 == 0) {
      return "Fizz";
    }
    else if (i % 5 == 0) {
      return "Buzz";
    }
    else {
      return to_string(i);
    }
}

TEST(FizzBuzzTest, TestOutputForOne) {
  EXPECT_EQ(fizzbuzz(1), "1");
}

TEST(FizzBuzzTest, TestOutputForThree) {
  EXPECT_EQ(fizzbuzz(3), "Fizz");
}

TEST(FizzBuzzTest, TestOutputForFive) {
  EXPECT_EQ(fizzbuzz(5), "Buzz");
}

TEST(FizzBuzzTest, TestOutputForFifteen) {
  EXPECT_EQ(fizzbuzz(15), "FizzBuzz");
}

File: fizzbuzz/python/fizzbuzz.py
# for loop based fizzbuzz
for i in range(1, 101):
  if i % 3 == 0 and i % 5 == 0:
    print('FizzBuzz\n')
  elif i % 3 == 0:
    print('Fizz\n')
  elif i % 5 == 0:
    print('Buzz\n')
  else:
    print(i, '\n')

# function based fizzbuzz for pytest
def fizzbuzz(num):
  if num > 100 or num <1:
    raise TypeError
  if num % 3 == 0 and num % 5 == 0:
    return 'FizzBuzz\n'
  elif num % 3 == 0:
    return 'Fizz\n'
  elif num % 5 == 0:
    return 'Buzz\n'
  else:
    return num
